StortSpelProjekt/Game/src/Gamefiles/Components/TeleportComponent.cpp

```diff
@@ -10,20 +10,20 @@ component::TeleportComponent::TeleportComponent(Entity* parent, Entity* player,
 {
 	m_NewSceneName = newSceneName;
 	m_pPlayerInstance = player;
-	EventBus::GetInstance().Subscribe(this, &TeleportComponent::OnCollision);
 }
 
 component::TeleportComponent::~TeleportComponent()
 {
-	EventBus::GetInstance().Unsubscribe(this, &TeleportComponent::OnCollision);
 }
 
 void component::TeleportComponent::OnInitScene()
 {
+	EventBus::GetInstance().Subscribe(this, &TeleportComponent::OnCollision);
 }
 
 void component::TeleportComponent::OnUnInitScene()
 {
+	EventBus::GetInstance().Unsubscribe(this, &TeleportComponent::OnCollision);
```

Comment:
Good! My bad for not doing this before :( I take full responsibility for my horrible and badly designed code :o